cmake/BLTMacros.cmake

```diff
@@ -209,19 +209,14 @@ macro(blt_add_library)
         #
         #  Header-only library support
         #
-        foreach (_file ${arg_HEADERS})
-            # Determine build location of headers
-            get_filename_component(_absolute ${_file} ABSOLUTE)
-            list(APPEND _build_headers ${_absolute})
-        endforeach()
-
-        #Note: This only works if both libraries are handled in the same directory,
-        #  otherwise just don't include non-header files in your source list.
-        set_source_files_properties(${_build_headers} PROPERTIES HEADER_FILE_ONLY ON)
-
-        add_library( ${arg_NAME} INTERFACE )
-        target_sources( ${arg_NAME} INTERFACE
-                        $<BUILD_INTERFACE:${_build_headers}>)
+        #  Note: Will not add header files to header-only interface libraries in older versions
+        #  of CMake when CMP0076 is set to NEW
+        #
+        if( ${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.19.0" )
+            add_library( ${arg_NAME} INTERFACE ${arg_HEADERS} )
+        else()
+            add_library( ${arg_NAME} INTERFACE )
```

Comment:
Do we want a warning for this case?

Reply:
I clarified the comment. Basically header only libraries don't need to add their sources but if you want them to show up in your IDE project then you have to. CMake is very fickle about this and depending on the version you have to jump through a lot of hoops. I went with the semi-sane answer to this in "newer" versions.

Reply:
We should note this in the RELEASE-NOTES as well, to give upgraders a heads-up